x-pack/plugins/watcher/server/models/action/slack_action.js

```diff
@@ -60,45 +62,50 @@ export class SlackAction extends BaseAction {
   // From Elasticsearch
   static fromUpstreamJson(json) {
     const props = super.getPropsFromUpstreamJson(json);
+    const { errors } = this.validateJson(json.actionJson);
+
+    Object.assign(props, {
+      to: json.actionJson.slack.message.to,
+      text: json.actionJson.slack.message.text
+    });
 
-    if (!json.actionJson.slack) {
-      throw badRequest(
-        i18n.translate('xpack.watcher.models.slackAction.absenceOfActionJsonSlackPropertyBadRequestMessage', {
+    const action = new SlackAction(props, errors);
+
+    return { action, errors };
+  }
+
+  static validateJson(json) {
+    const errors = [];
+
+    if (!json.slack) {
+      errors.push({
+        code: ERROR_CODES.ERR_PROP_MISSING,
+        message: i18n.translate('xpack.watcher.models.slackAction.absenceOfActionJsonSlackPropertyBadRequestMessage', {
```

Comment:
`absenceOfActionJsonSlackPropertyBadRequestMessage` is a bit weird, maybe `actionJsonSlackPropertyMissingBadRequestMessage` ?

Reply:
No problem to change these but then we would need to review all the other keys for Watch as they follow the pattern `absenceOf...`. What do you think?